pennylane/capture/dynamic_shapes.py

```diff
@@ -113,3 +115,82 @@ def f(n):
         return None, ()
     abstracted_axes = jax.tree_util.tree_unflatten(structure, abstracted_axes)
     return abstracted_axes, abstract_shapes
+
+
+def register_custom_staging_rule(
+    primitive, get_outvars_from_params: Callable[[dict], list["jax.core.Var"]]
+) -> None:
+    """Register a custom staging rule for a primitive, where the output should match the variables retrieved by
+    ``get_outvars_from_params``.
+
+    Args:
+        primitive (jax.core.Primitive): a jax primitive we want to register a custom staging rule for
+        get_outvars_from_params (Callable[[dict], list[jax.core.Var]]): A function that takes in the equation's ``params``
+            and returns ``jax.core.Var`` we need to mimic for the primitives return.
+
+    For example, the ``cond_prim`` will request its custom staging rule like:
+
+    .. code-block:: python
+
+        register_custom_staging_rule(cond_prim, lambda params: params['jaxpr_branches'][0].outvars)
+
+    The return of any ``cond_prim`` will match the output variables of the first jaxpr branch.
+
+    """
+    # see https://github.com/jax-ml/jax/blob/9e62994bce7c7fcbb2f6a50c9ef89526cd2c2be6/jax/_src/lax/lax.py#L3538
+    # and https://github.com/jax-ml/jax/blob/9e62994bce7c7fcbb2f6a50c9ef89526cd2c2be6/jax/_src/lax/lax.py#L208
+    # for reference to how jax is handling staging rules for dynamic shapes in v0.4.28
+    # see also capture/intro_to_dynamic_shapes.md
+
+    def _tracer_and_outvar(
+        jaxpr_trace: pe.DynamicJaxprTrace,
+        outvar: jax.core.Var,
+        env: dict[jax.core.Var, jax.core.Var],
+    ) -> tuple[pe.DynamicJaxprTracer, jax.core.Var]:
+        """
+        Create a new tracer and returned var from the true branch outvar
+        returned vars are cached in env for use in future shapes
```

Comment:
```suggestion
        Create a new tracer and return var from the true branch outvar.
        Returned vars are cached in env for use in future shapes
```
please let me know if I misunderstood the meaning